File: src/c/sequence/sequence_without_array.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <limits.h>

#define ROUNDS INT_MAX
#define RED 1
#define BLACK -1
#define ZERO 0

char getNumber() {
    
    char number;
    number = rand() % 37;
        if (number != 0) {
            number = (number % 2) * 2 - 1;
        }
    return number;
}

int main() {

    time_t t;
    srand(time(&t));

	//define variables
	int zeroes, blacks, reds, maxZeroSequence, maxRedSequence, maxBlackSequence;
	zeroes = blacks = reds = maxZeroSequence = maxRedSequence = maxBlackSequence = 0;

	//Fill sequence
	
	char container = getNumber();
	
    //Init counter
    int count = 0;

    //Count sequences
    for (int i = 0; i < ROUNDS; i++) {

        //Count red sequence
        if (container == RED) {
            while (container == RED) {
                reds++;
                count++;
                if (++i == ROUNDS) break;
                container = getNumber();
            }
            if (count > maxRedSequence) {
                maxRedSequence = count;
            }

        //Count black sequence
        } else if (container == BLACK) {
            while (container == BLACK) {
                blacks++;
                count++;
                if (++i == ROUNDS) break;
                container = getNumber();
            }
            if (count > maxBlackSequence) {
                maxBlackSequence = count;
            }

        //count zero sequence
        } else if (container == ZERO) {
            while (container == ZERO) {
                zeroes++;
                count++;
                if (++i == ROUNDS) break;
                container = getNumber();
            }
            if (count > maxZeroSequence) {
                maxZeroSequence = count;
            }
        }
        count = 0;
        i--;
    }

    printf("Reds:\t\t\t%d\nMaximum sequence:\t%d\n\nBlacks:\t\t\t%d\nMaximum sequence:\t%d\n\nZeroes:\t\t\t%d\nMaximum sequence:\t%d\n", reds, maxRedSequence, blacks, maxBlackSequence, zeroes, maxZeroSequence);

	return 0;
}

File: src/python/sequence/sequence_without_array.py
import random

def getNumber():
    num = random.randint(0, 36)

    if num != 0:
        num = (num % 2) * 2 - 1

    return num

#Define constants
ROUNDS = 1000
RED = 1
BLACK = -1
ZERO = 0

#init variables
zeroes = 0
blacks = 0
reds = 0
maxZeroSequence = 0
maxRedSequence = 0
maxBlackSequence = 0

container = getNumber()

#Init counter
count = 0

#Count sequences
for i in range(ROUNDS):

    #Count red sequence
    if container == RED:
        while container == RED:
            reds += 1
            count += 1
            i += 1
            if i == ROUNDS:
                break
            container = getNumber()
        
            if count > maxRedSequence:
                maxRedSequence = count

    #Count black sequence
    elif container == BLACK:
        while container == BLACK:
            blacks += 1
            count += 1
            i += 1
            if i == ROUNDS:
                break
            container = getNumber()
        
        if count > maxBlackSequence:
            maxBlackSequence = count

    #count zero sequence
    elif container == ZERO:
        while container == ZERO:
            zeroes += 1
            count += 1
            i += 1
            if i == ROUNDS:
                break
            container = getNumber()
        
        if count > maxZeroSequence:
            maxZeroSequence = count
        
    count = 0
    i -= 1
    
#Result output
text = "-----------------------------------\n"

text += "Reds:\t\t\t{}\n".format(str(reds))
text += "Maximum sequence: \t{}\n\n".format(str(maxRedSequence))

text += "Blacks:\t\t\t{}\n".format(str(blacks))
text += "Maximum sequence: \t{}\n\n".format(str(maxBlackSequence))

text += "Zeroes:\t\t\t{}\n".format(str(zeroes))
text += "Maximum sequence: \t{}\n\n".format(str(maxZeroSequence))

print(text)
